src/runtime-rs/crates/resource/src/rootfs/nydus_rootfs.rs

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+use std::{fs, sync::Arc};
+
+use super::{Rootfs, TYPE_OVERLAY_FS};
+use crate::{
+    rootfs::{HYBRID_ROOTFS_WRITABLE_LAYER_LOWER_DIR, ROOTFS},
+    share_fs::{
+        do_get_guest_path, do_get_guest_share_path, get_host_rw_shared_path, rafs_mount,
+        ShareFsMount, ShareFsRootfsConfig, PASSTHROUGH_FS_DIR,
+    },
+};
+use agent::Storage;
+use anyhow::{anyhow, Context, Result};
+use async_trait::async_trait;
+use hypervisor::Hypervisor;
+use kata_types::mount::{Mount, NydusExtraOptions};
+
+// Used for nydus rootfs
+pub(crate) const NYDUS_ROOTFS_TYPE: &str = "fuse.nydus-overlayfs";
+// Used for Nydus v5 rootfs version
+const NYDUS_ROOTFS_V5: &str = "v5";
+// Used for Nydus v6 rootfs version
+const NYDUS_ROOTFS_V6: &str = "v6";
+
+const SNAPSHOT_DIR: &str = "snapshotdir";
+const KATA_OVERLAY_DEV_TYPE: &str = "overlayfs";
+
+pub(crate) struct NydusRootfs {
+    guest_path: String,
+    rootfs: Storage,
+}
+
+impl NydusRootfs {
+    pub async fn new(
+        share_fs_mount: &Arc<dyn ShareFsMount>,
+        h: &dyn Hypervisor,
+        sid: &str,
+        cid: &str,
+        rootfs: &Mount,
+    ) -> Result<Self> {
+        let extra_options =
+            NydusExtraOptions::new(rootfs).context("failed to parse nydus extra options")?;
+        info!(sl!(), "extra_option {:?}", &extra_options);
+        let rafs_meta = &extra_options.source;
+        let (rootfs_storage, rootfs_guest_path) = match extra_options.fs_version.as_str() {
+            NYDUS_ROOTFS_V5 => {
+                let rafs_mnt =
+                    do_get_guest_share_path(HYBRID_ROOTFS_WRITABLE_LAYER_LOWER_DIR, cid, true);
+                rafs_mount(
+                    h,
+                    rafs_meta.to_string(),
+                    rafs_mnt,
+                    extra_options.config.clone(),
+                    None,
+                )
+                .await
+                .context("failed to do rafs mount")?;
+                let container_share_dir = get_host_rw_shared_path(sid)
+                    .join(PASSTHROUGH_FS_DIR)
+                    .join(cid);
+                let rootfs_dir = container_share_dir.join(ROOTFS);
+                fs::create_dir_all(rootfs_dir).context("failed to create directory")?;
+                let mount_result = share_fs_mount
+                    .share_rootfs(ShareFsRootfsConfig {
+                        cid: cid.to_string(),
+                        source: extra_options.snapshot_dir.clone(),
+                        target: SNAPSHOT_DIR.to_string(),
+                        readonly: true,
+                        is_rafs: false,
```

Comment:
`is_rafs` should be true or false?

Reply:
The read-only files will be put under the rafs directory, so the metadata of nydus rootfs is under it. The snapshot dir is used as the upper layer which is wirtable, so it don't need to be under rafs

Reply:
I asked this question because I couldn't find any place that set `is_rafs` to true. Did I missed some places?